node/src/manager/commands/fix_block.rs

```diff
@@ -0,0 +1,300 @@
+        Ok(())
+    }
+
+    /// Queries the [`ChainStore`] about the chain head.
+    pub(super) fn find_chain_head(chain_store: &ChainStore) -> anyhow::Result<i32> {
+        let chain_head: Option<i32> = chain_store.chain_head_block(&chain_store.chain)?;
+        chain_head.ok_or_else(|| anyhow!("Could not find the chain head for {}", chain_store.chain))
+    }
+}
+
+mod helpers {
+    use super::*;
+    use graph::prelude::hex;
+    use std::io::{self, Write};
+
+    /// Tries to parse a [`H256`] from a hex string.
+    pub(super) fn parse_block_hash(hash: &str) -> anyhow::Result<H256> {
+        let hash = hash.trim_start_matches("0x");
+        let hash = hex::decode(hash)
+            .with_context(|| format!("Cannot parse H256 value from string `{}`", hash))?;
+        Ok(H256::from_slice(&hash))
+    }
+
+    /// Asks users if they are certain about truncating the whole block cache.
+    pub(super) fn prompt_for_confirmation() -> anyhow::Result<bool> {
+        print!("This will delete all cached blocks.\nProceed? [y/N] ");
+        io::stdout().flush()?;
+
+        let mut answer = String::new();
+        io::stdin().read_line(&mut answer)?;
+        answer.make_ascii_lowercase();
+
+        match answer.trim() {
+            "y" | "yes" => Ok(true),
+            _ => Ok(false),
+        }
+    }
+
+    /// Convenience function for extracting values from unary sets.
+    pub(super) fn get_single_item<I, T>(name: &'static str, collection: I) -> anyhow::Result<T>
+    where
+        I: IntoIterator<Item = T>,
+    {
+        let mut iterator = collection.into_iter();
+        match (iterator.next(), iterator.next()) {
+            (Some(a), None) => Ok(a),
+            (None, None) => bail!("Expected a single {name} but found none."),
+            _ => bail!("Expected a single {name} but found multiple occurrences."),
+        }
+    }
+}
+
+/// Custom range type that supports being parsed from a string.
+mod ranges {
+    use graph::prelude::anyhow::{self, bail};
+    use std::str::FromStr;
+
+    pub(super) struct Range {
+        pub(super) lower_bound: Option<i32>,
+        pub(super) upper_bound: Option<i32>,
+        pub(super) inclusive: bool,
+    }
+
+    impl Range {
+        fn new(lower_bound: Option<i32>, upper_bound: Option<i32>, inclusive: bool) -> Self {
+            Self {
+                lower_bound,
+                upper_bound,
+                inclusive,
+            }
+        }
+
+        pub(super) fn min_max(&self) -> anyhow::Result<(i32, Option<i32>)> {
+            let min = match self.lower_bound {
+                None | Some(0) => 1, // Never include the genesis block
+                Some(x) if x < 0 => anyhow::bail!("Negative block number"),
+                Some(x) => x,
+            };
+            let inclusive = if self.inclusive { 1 } else { 0 };
+            let max = self.upper_bound.map(|x| x + inclusive);
+            Ok((min, max))
+        }
+    }
+
+    impl FromStr for Range {
```

Comment:
I would avoid all this and just have users pass in `--from` and `--to` (inclusive) options.